### PR TITLE
Support multiple alternative locations for azurerm_api_management

### LINKED ISSUE #4171: Unable to define multiple `additional_location` on `azurerm_api_management`
**Terraform (and AzureRM Provider) Version**

- TF 0.11.14
- AzureRM 1.33.0

**Affected Resource(s)**

- azurerm_api_management

**Terraform Configuration**

```
resource "azurerm_api_management" "apim" {
  name                = "${var.name}"
  location            = "${var.location}"
  resource_group_name = "${var.rg_name}"

  sku = ["${var.sku}"]

  additional_location {
    location = "brazilsouth"
  }
  additional_location {
    location = "eastus2"
  }
  additional_location {
    location = "southeastasia"
  }

  [...]

  tags = "${var.tags}"
}
```

**Expected Behavior**

Enable several additional_location on the Apim resource

**Actual Behavior**

"Error: module.api_management.azurerm_api_management.apim: additional_location: attribute supports 1 item maximum, config has 3 declared"

On the Terraform documentation we can find that additional_location can be one or more blocks : 
> additional_location - One or more additional_location blocks as documented below.

By searching in the code, I found this : 

https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_api_management.go#L123

```
"additional_location": {
                Type:     schema.TypeList,
                Optional: true,
                MaxItems: 1,
[...]
}
```

This MaxItems variable should be removed to handle multiple additional_location I think.

### PR DESCRIPTION
**Pull request by pdecat**
Resolves #4171

### CODE REVIEW

**Comment by pdecat** [comment]
This changes allows any number of alternative locations.

It is just the acceptance that tests with two alternative locations.

**Comment by pdecat** [comment]
Acceptance test result:

```
# make testacc TEST=./azurerm TESTARGS='-run=TestAccAzureRMApiManagement_complete'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./azurerm -v -run=TestAccAzureRMApiManagement_complete -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMApiManagement_complete
=== PAUSE TestAccAzureRMApiManagement_complete
=== CONT  TestAccAzureRMApiManagement_complete
--- PASS: TestAccAzureRMApiManagement_complete (1571.92s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm       1571.951s
```

File: azurerm/provider_test.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func testAccPreCheck(t *testing.T) {
 		"ARM_TENANT_ID",
 		"ARM_TEST_LOCATION",
 		"ARM_TEST_LOCATION_ALT",
+		"ARM_TEST_LOCATION_ALT2",
```
Comment (tombuildsstuff) [inline azurerm/provider_test.go]: internal: I've configured this in CI

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-08-29 at 15 57 08

**Comment by pdecat** [comment]
Great, thanks @tombuildsstuff!

**Pull request merged**